modules/ding_content_rss/ding_content_rss.module

```diff
@@ -0,0 +1,216 @@
+      }
+      elseif ($view->name == 'ding_content_news_rss') {
+        $view->result[$key]->field_field_ding_news_body[0]['rendered'] = preg_replace('#http://([\w\.])+(/ting)/(.+)#Ui', 'bappsp://$3', $result->field_field_ding_news_body[0]['rendered']);
+      }
+    }
+  }
+}
+
+/**
+ * Implements hook_field_formatter_info().
+ */
+function ding_content_rss_field_formatter_info() {
+  return array(
+    'ding_content_rss_location' => array(
+      'label' => t('RSS friendly location'),
+      'field types' => array('addressfield'),
+      'description' => t('Output address information on a single HTML stripped line.'),
+    ),
+    'ding_content_rss_date_time' => array(
+      'label' => t('RFC-2822 formatted date'),
+      'field types' => array('datetime'),
+      'description' => t('Format date and time as RFC-2822 date/time string.'),
+    ),
+  );
+}
+
+/**
+ * Implements hook_field_formatter_view().
+ */
+function ding_content_rss_field_formatter_view($entity_type, $entity, $field, $instance, $langcode, $items, $display) {
+  $element = array();
+  switch ($display['type']) {
+    case 'ding_content_rss_date_time':
+      foreach ($items as $delta => $item) {
+        $element[$delta] = array(
+          '#markup' => strtotime($item[$display['settings']['fromto']] . ' ' . $item['timezone_db']),
+        );
+      }
+      break;
+
+    case 'ding_content_rss_location':
+      foreach ($items as $delta => $item) {
+        $location = array();
+        if (!empty($item['name_line'])) {
+          foreach ($item as $part) {
+            if (!empty($part)) {
+              $location[] = $part;
+            }
+          }
+        }
+        else {
+          // Try to get library title as location.
+          $wrapper = entity_metadata_wrapper('node', $entity);
+          $libraries = $wrapper->og_group_ref->value();
+          foreach ($libraries as $library) {
+            $location[] = $library->title;
+          }
+        }
+
+        // Fallback to groups title for location.
+        if (count($location) == 0) {
+          $wrapper = entity_metadata_wrapper('node', $entity);
+          $groups = $wrapper->ding_event_groups_ref->value();
+          foreach ($groups as $group) {
+            $location[] = $group->title;
+          }
+        }
+
+        $element[$delta] = array(
+          '#markup' => implode(', ', array_reverse($location)),
+        );
+      }
+      break;
+  }
+  return $element;
+}
+
+/**
+ * Implements hook_field_formatter_settings_form().
+ */
+function ding_content_rss_field_formatter_settings_form($field, $instance, $view_mode, $form, &$form_state) {
+  $display = $instance['display'][$view_mode];
+  $settings = $display['settings'];
+  $form = array();
+  $form['fromto'] = array(
+    '#title' => t('Display:'),
+    '#type' => 'select',
+    '#options' => array(
+      'both' => t('Both Start and End dates'),
```

Comment:
I might be missing something here, but https://github.com/ding2/ding2/pull/1010/files#diff-4518520246760e0297f29fbeb0664724R54 only runs strtotime on a single value?